src/new/wallet.js

```diff
@@ -184,6 +184,182 @@ class HathorWallet extends EventEmitter {
     return historyTransactions;
   }
 
+  /**
+   * Get utxos of the wallet addresses
+   *
+   * @typedef {Object} UtxoOptions
+   * @property {number} max_utxos - Maximum number of utxos to aggregate. Default to MAX_INPUTS (255).
+   * @property {string} token - Token to filter the utxos. If not sent, we select only HTR utxos.
+   * @property {string} filter_address - Address to filter the utxos.
+   * @property {number} amount_smaller_than - Maximum limit of utxo amount to filter the utxos list. We will consolidate only utxos that have an amount lower than this value. Integer representation of decimals, i.e. 100 = 1.00.
+   * @property {number} amount_bigger_than - Minimum limit of utxo amount to filter the utxos list. We will consolidate only utxos that have an amount bigger than this value. Integer representation of decimals, i.e. 100 = 1.00.
+   * @property {number} maximum_amount - Limit the maximum total amount to consolidate summing all utxos. Integer representation of decimals, i.e. 100 = 1.00.
+   * @property {boolean} only_available_utxos - Use only available utxos (not locked)
+   *
+   * @typedef {Object} UtxoDetails
+   * @property {number} total_amount_available - Maximum number of utxos to aggregate. Default to MAX_INPUTS (255).
+   * @property {number} total_utxos_available - Token to filter the utxos. If not sent, we select only HTR utxos.
+   * @property {number} total_amount_locked - Address to filter the utxos.
+   * @property {number} total_utxos_locked - Maximum limit of utxo amount to filter the utxos list. We will consolidate only utxos that have an amount lower than this value. Integer representation of decimals, i.e. 100 = 1.00.
+   * @property {{ address: string, amount: number, tx_id: string, locked: boolean, index: number }[]} utxos - Array of utxos
+   *
+   * @param {UtxoOptions} options Utxo filtering options
+   *
+   * @return {UtxoDetails} Utxos and meta information about it
+   *
+   */
+  getUtxos(options = {}) {
+    storage.setStore(this.store);
+    const historyTransactions = Object.values(this.getTxHistory());
+    const utxoDetails = {
+      total_amount_available: 0,
+      total_utxos_available: 0,
+      total_amount_locked: 0,
+      total_utxos_locked: 0,
+      utxos: [],
+    };
+
+    // Iterate through transactions
+    for (let i = 0; i < historyTransactions.length; i++) {
+      const transaction = historyTransactions[i];
+
+      // Voided transactions should be ignored
+      if (transaction.is_voided) continue;
+
+      // Iterate through outputs
+      for (let j = 0; j < transaction.outputs.length; j++) {
+        const output = transaction.outputs[j];
+
+        const is_unspent = output.spent_by === null;
+        const locked = !wallet.canUseUnspentTx(output, j);
```

Comment:
This second parameter is the transaction height, in case of a block. Why is it `j` here?

Reply:
Fixed. Now I'm using `transaction.height`.